visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -1,15 +1,62 @@
 import { CodeMapNode } from "../codeCharta.model"
+import * as codeCharta from "../../../package.json"
+import { DialogService } from "../ui/dialog/dialog.service"
 
 export class FileValidator {
+	public static validate(file: { nodes: CodeMapNode[] }, dialogService?: DialogService): string[] {
```

Comment:
Can you please refactor this method? Its really long. Also is there a significant difference of the two different exclamation icons? I would just use one and if it is always added before one array item, we could just add it together with the "<br>" inside the `fileChooser.component`

Reply:
The problem is that in case of the users Minor API version is wrong we display a warning, but continue rendering, the current implementation of codeCharta.service does not support warnings

Reply:
we just stop rendering if any output other then 0 is returned
But maybe I should change that

Reply:
I am not getting the problem, sorry 😁 how about returning an object like the following and then the fileChooser component can call the correct dialogs? 
```
{
  warnings: [],
  errors: []
}
```

Reply:
sounds good, will need to address codeCharta and its test accordingly, 
I will do it, thank you